src/components/TodoHeader/TodoHeader.tsx

```diff
@@ -0,0 +1,135 @@
+import {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import classnames from 'classnames';
+
+import { TodoContext } from '../../context/TodoContext';
+import { ErrorContext } from '../../context/ErrorContext';
+import { TodoTempContext } from '../../context/TodoTempContext';
+import { USER_ID } from '../../utils/variables';
+import { createTodo } from '../../api/todos';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  onHandleActive: (value: boolean) => void;
+};
+
+export const TodoHeader: React.FC<Props> = ({ onHandleActive }) => {
+  const { todos, setTodos } = useContext(TodoContext);
+  const { setTodoTemp } = useContext(TodoTempContext);
+  const { errorMessage, setErrorMessage } = useContext(ErrorContext);
+
+  const [title, setTitle] = useState('');
+  const [isInputDisabled, setIsInputDisabled] = useState(false);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const isActiveItemsLeft = todos.filter(({ completed }) => !completed).length;
+
+  useEffect(() => {
+    inputRef?.current?.focus();
+  }, [todos, errorMessage]);
+
+  const toggleAll = () => {
+    const allCompleted = todos.every(({ completed }) => completed);
+
+    if (allCompleted) {
+      setTodos(currentTodos => {
+        return currentTodos.map(todo => ({
+          ...todo,
+          completed: false,
+        }));
+      });
+    } else {
+      setTodos(currentTodos => {
+        return currentTodos.map(todo => ({
+          ...todo,
+          completed: true,
+        }));
+      });
+    }
+  };
```

Comment:
It should also change the status of todos in the database, as when the user reloads the page todos have the same status as previously.